### ISSUE TITLE
hledger-web: register's date links are wrong with multiple input files

### ISSUE THREAD

**Issue by amigalemming** [simonmichael/hledger#1041]
In `http://localhost:5000/register?q=inacct%3A"MyAccount"`
I can click on a transaction date.
This gets me to, say, `http://localhost:5000/journal#transaction-167`
but the fragment `transaction-167` has the wrong number.
I am using multiple input files and the transactions are not strictly ordered by date.
I should prepare a small example, but maybe you can already confirm the problem with your data?

**Comment by simonmichael** [comment]
@amigalemming good catch, thanks for the report. Yes, I confirm that the register's hyperlinked dates link to the wrong transaction ids if the journal entries are not in date order, eg with examples/sample.journal disordered like so:
```
2008/12/31 * pay off
    liabilities:debts  $1
    assets:bank:checking

2008/06/01 gift
    assets:bank:checking  $1
    income:gifts

2008/01/01 income
    assets:bank:checking  $1
    income:salary

2008/06/03 * eat & shop
    expenses:food      $1
    expenses:supplies  $1
    assets:cash

2008/06/02 save
    assets:bank:saving  $1
    assets:bank:checking
```

**Comment by simonmichael** [comment]
Well actually no.. the ids look correct to me, both in the register's hyperlinks and in the journal view's anchors. Are you sure both journal and register are up to date in your browser and being served by the same hledger-web instance ? If so please do show a small example.

**Comment by amigalemming** [comment]
On Sun, 9 Jun 2019, Simon Michael wrote:

> @amigalemming good catch, thanks for the report. Yes, I confirm that the register's hyperlinked dates link to the
> wrong transaction ids if the journal entries are not in date order, eg with examples/sample.journal disordered
> like so:
> 
> 2008/12/31 * pay off
>     liabilities:debts  $1
>     assets:bank:checking
> 
> 2008/06/01 gift
>     assets:bank:checking  $1
>     income:gifts
> 
> 2008/01/01 income
>     assets:bank:checking  $1
>     income:salary
> 
> 2008/06/03 * eat & shop
>     expenses:food      $1
>     expenses:supplies  $1
>     assets:cash
> 
> 2008/06/02 save
>     assets:bank:saving  $1
>     assets:bank:checking


The problem seems actually be caused by using two journal files. Duplicate 
your example journal and replace 2008 by 2009, then start the web server 
using both journals. You see that the dates for corresponding transactions 
in 2008 and 2009 point to the same transaction number.

I guess the transaction numbers should be extended by journal numbers.

**Comment by simonmichael** [comment]
Fixed by #1242.
